providers/aws/resources/shared.go

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resources
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	"github.com/aws/smithy-go/transport/http"
+	"go.mondoo.com/cnquery/llx"
+	"go.mondoo.com/cnquery/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/providers/aws/connection"
+	"go.mondoo.com/cnquery/providers/network/resources/certificates"
+	"go.mondoo.com/cnquery/types"
+	"k8s.io/client-go/util/cert"
+)
+
+const (
+	elbv1LbArnPattern           = "arn:aws:elasticloadbalancing:%s:%s:loadbalancer/classic/%s"
+	cloudwatchAlarmArnPattern   = "arn:aws:cloudwatch:%s:%s:metricalarm/%s/%s"
+	ec2InstanceArnPattern       = "arn:aws:ec2:%s:%s:instance/%s"
+	securityGroupArnPattern     = "arn:aws:ec2:%s:%s:security-group/%s"
+	volumeArnPattern            = "arn:aws:ec2:%s:%s:volume/%s"
+	snapshotArnPattern          = "arn:aws:ec2:%s:%s:snapshot/%s"
+	internetGwArnPattern        = "arn:aws:ec2:%s:%s:gateway/%s"
+	vpnConnArnPattern           = "arn:aws:ec2:%s:%s:vpn-connection/%s"
+	networkAclArnPattern        = "arn:aws:ec2:%s:%s:network-acl/%s"
+	imageArnPattern             = "arn:aws:ec2:%s:%s:image/%s"
+	keypairArnPattern           = "arn:aws:ec2:%s:%s:keypair/%s"
+	s3ArnPattern                = "arn:aws:s3:::%s"
+	dynamoTableArnPattern       = "arn:aws:dynamodb:%s:%s:table/%s"
+	limitsArn                   = "arn:aws:dynamodb:%s:%s"
+	dynamoGlobalTableArnPattern = "arn:aws:dynamodb:-:%s:globaltable/%s"
+	rdsInstanceArnPattern       = "arn:aws:rds:%s:%s:db:%s"
+	apiArnPattern               = "arn:aws:apigateway:%s:%s::/apis/%s"
+	apiStageArnPattern          = "arn:aws:apigateway:%s:%s::/apis/%s/stages/%s"
+)
+
+func (a *mqlAws) regions() ([]interface{}, error) {
+	conn := a.MqlRuntime.Connection.(*connection.AwsConnection)
+	res := []interface{}{}
+	regions, err := conn.Regions()
+	for i := range regions {
+		res = append(res, regions[i])
+	}
+	return res, err
+}
+
+func Is400AccessDeniedError(err error) bool {
+	var respErr *http.ResponseError
+	if errors.As(err, &respErr) {
+		if respErr.HTTPStatusCode() == 400 && strings.Contains(respErr.Error(), "AccessDeniedException") {
+			return true
+		}
+	}
+	return false
+}
+
+func toBool(b *bool) bool {
```

Comment:
For all these `toX` funcs, there should already be one in the `convert` package or llx has also `XDataPtr` versions which might make it even simpler.

Reply:
oh nice!